core/src/main/java/fr/sncf/osrd/envelope/OverlayEnvelopePartBuilder.java

```diff
@@ -2,14 +2,21 @@
 
 import static fr.sncf.osrd.envelope.EnvelopeCursor.NextStepResult.NEXT_PART;
 import static fr.sncf.osrd.envelope.EnvelopeCursor.NextStepResult.NEXT_REACHED_END;
+import static fr.sncf.osrd.envelope.EnvelopePhysics.interpolateStepTime;
+import static fr.sncf.osrd.envelope.EnvelopePhysics.intersectStepWithSpeed;
 
-import fr.sncf.osrd.utils.CompareSign;
+import fr.sncf.osrd.utils.CmpOperator;
 
 /** Creates an envelope part which always keeps a speed lower than a given envelope */
 public class OverlayEnvelopePartBuilder implements StepConsumer {
     /** This cursor is updated as the overlay is built. It must not be modified elsewhere until build is called */
     public final EnvelopeCursor cursor;
 
+    /** The maximum / minimum speed allowed for this overlay, or NaN. */
+    private double speedThreshold = Double.NaN;
+    /** How to compare the overlay speed to the threshold speed. */
+    private CmpOperator speedThresholdOperator = null;
+
```

Comment:
This is only used in tests, I assume it will be used in construction margins eventually?

Reply:
indeed